tests/test_nucname.py

```diff
@@ -717,10 +717,10 @@ def test_mcnp_to_id():
     for val, id in set(zip(vals, caseids)):
         if val is None:
             continue
-        yield check_cases, nucname.mcnp_to_id, val, id
+        check_cases(nucname.mcnp_to_id, val, id)
```

Comment:
parametrize?

Reply:
Question: The next 6-8 functions that were yield-based tests have an `if` statement that skipped testing the value if the value is `None`. I have rewritten it to only check the case if the value is not `None`. It also seems that in some tests there is no value `None` so I am wondering if we should keep the `if` statement for these tests? and what the value of adding `None` to a test suite is if we are skipping it anyway?

Reply:
I see what you are saying.  I wonder if we can setup the parameters differently??

In each case, it is matching some formatted version of a nuclide name (in `vals`) with its canonical version (in `caseids`).  

What about 
```
set( (val, casid) in zip([...], caseids) if val not None )
```
or something like this..?? not sure if that will work.

This will strip out the `None` values from `vals` after zipping them together with `caseids`.

Reply:
I wonder if it would work to make a function that did this that you use in the `parametrize()`?
```
def build_params(vals, caseids):
    params = zip(vals, caseids)
    params = set( (val, caseid) for val, caseid in params if val not None)
    return params

@pytest.mark.parametrize("val, id", build_params([...], caseids))
def fucn():
```